eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentEvents.scala

```diff
@@ -232,6 +235,31 @@ object PaymentFailure {
 
 }
 
+case class PaymentFailureSummary(amount: MilliSatoshi, route: Seq[PublicKey], message: String)
+
+object PaymentFailureSummary {
+  def apply(failure: PaymentFailure): PaymentFailureSummary = {
+    val route = failure.route.map(_.nodeId) ++ failure.route.lastOption.map(_.nextNodeId)
+    val message = failure match {
+      case LocalFailure(_, _, t) => t.getMessage
+      case RemoteFailure(_, _, Sphinx.DecryptedFailurePacket(origin, failureMessage)) => s"$origin returned: ${failureMessage.message}"
+      case _: UnreadableRemoteFailure => "unreadable remote failure"
+    }
+    PaymentFailureSummary(failure.amount, route, message)
+  }
+}
+
+case class PaymentFailedSummary(totalAmount: MilliSatoshi, pathFindingExperiment: String, failures: Seq[PaymentFailureSummary]) {
```

Comment:
Why not include the whole config? We can always filter out the relevant attributes in the serializer.

```suggestion
case class PaymentFailedSummary(config: SendPaymentConfig, failures: Seq[PaymentFailureSummary]) {
```

Reply:
Meh, it doesn't really contain the interesting bits, more than half of the fields of `SendPaymentConfig` are things we really don't care about

Reply:
True, but selecting the relevant fields could be considered a serialization detail. For example, in 4bab67ebdafa6a92bcc984178ce8f836233a583f I added the destination node id and had to update the definition of `PaymentFailedSummary`.

Reply:
Right, why not in that case.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1966/commits/eb9f63309a905c81e3ef31d30b76302495af0d72